src/LanguageServer/Protocol/Handler/Completion/CompletionResultFactory.cs

```diff
@@ -86,7 +86,9 @@ internal static class CompletionResultFactory
                 ItemDefaults = new LSP.CompletionListItemDefaults
                 {
                     EditRange = capabilityHelper.SupportDefaultEditRange ? ProtocolConversions.TextSpanToRange(defaultSpan, documentText) : null,
-                    Data = capabilityHelper.SupportCompletionListData ? resolveData : null
+                    Data = capabilityHelper.SupportCompletionListData ? resolveData : null,
+                    // By default, all text edits we create include the appropriate whitespace, so tell the client to leave it as-is.
+                    InsertTextMode = LSP.InsertTextMode.AsIs,
```

Comment:
>InsertTextMode [](http://example.com/codeflow?start=20&length=14)

Would it make sense to conditionally set this based on whether the capabilityHelper indicated insertTextModeSupport is supported?

Reply:
accidentally force pushed, but fixed